tests/lib/rules/no-console.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * @fileoverview Tests for rule "no-console"
+ * @author Pig Fang <g-plane@hotmail.com>
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const rule = require("../../../lib/rules/no-console");
+const RuleTester = require("eslint").RuleTester;
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester();
+
+ruleTester.run("no-console", rule, {
+    valid: [
+        "console;",
+        "if (true) console;"
+    ],
+    invalid: [
+        {
+            code: "console.log",
+            output: "",
+            errors: [{ messageId: "unexpected", type: "MemberExpression" }]
+        },
+        {
+            code: "console.log()",
+            output: "",
+            errors: [{ messageId: "unexpected", type: "MemberExpression" }]
+        },
+        {
+            code: "if (true) console.log",
+            output: "if (true) {}",
+            errors: [{ messageId: "unexpected", type: "MemberExpression" }]
+        },
+        {
+            code: "if (true) { console.log }",
+            output: "if (true) {  }",
+            errors: [{ messageId: "unexpected", type: "MemberExpression" }]
+        },
+        {
+            code: "if (true) console.log()",
```

Comment:
we might need to check the arguments:
```js
console.log(foo); // safe to remove
console.log(foo, bar): // safe to remove
console.log("foo"); // safe to remove
console.log(`foo`); // safe to remove

console.log(foo()); // unsafe to remove
console.log(`${foo()}`); unsafe to remove // unsafe to remove
```

btw, mind adding something like:
```js
let log = console.log;
```

Reply:
So should we remove that when the arguments only contain literal or identifier?

Reply:
I guess so, since we don't know if the others have side effects.